hack/tools/fetch-images/main.go

```diff
@@ -0,0 +1,268 @@
+// Copyright 2024 Nutanix. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"slices"
+	"strings"
+	"text/template"
+
+	"github.com/d2iq-labs/helm-list-images/pkg"
+	"github.com/d2iq-labs/helm-list-images/pkg/k8s"
+	yamlv2 "gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+const (
+	createImagesCMD          = "create-images"
+	defaultHelmAddonFilename = "values-template.yaml"
+)
+
+type ChartInfo struct {
+	repo         string
+	name         string
+	valuesFile   string
+	stringValues []string
+}
+
+func main() {
+	args := os.Args
+	var (
+		chartDirectory     string
+		helmChartConfigMap string
+		carenVersion       string
+	)
+	flagSet := flag.NewFlagSet(createImagesCMD, flag.ExitOnError)
+	flagSet.StringVar(&chartDirectory, "chart-directory", "",
+		"path to chart directory for CAREN")
+	flagSet.StringVar(&helmChartConfigMap, "helm-chart-configmap", "",
+		"path to chart directory for CAREN")
+	flagSet.StringVar(&carenVersion, "caren-version", "",
+		"caren version for images override")
+	err := flagSet.Parse(args[1:])
+	if err != nil {
+		fmt.Println("failed to parse args", err.Error())
+		os.Exit(1)
+	}
+	chartDirectory, err = EnsureFullPath(chartDirectory)
+	if err != nil {
+		fmt.Println("failed to get full path for chart", err.Error())
+		os.Exit(1)
+	}
+	helmChartConfigMap, err = EnsureFullPath(helmChartConfigMap)
+	if err != nil {
+		fmt.Println("failed to get full path for helmChartConfigMap", err.Error())
+		os.Exit(1)
+	}
+	if chartDirectory == "" || helmChartConfigMap == "" {
+		fmt.Println("chart-directory helm-chart-configmap must be set")
+		os.Exit(1)
+	}
+	i := &ChartInfo{
+		name: chartDirectory,
+	}
+	if carenVersion != "" {
```

Comment:
should make carenVersion required flag.

Reply:
it gets run by the make target so i think it will be ok